### PR TITLE
ATMOSPHERIS! This time with more emotion

### PR DESCRIPTION
**Pull request by Dissentbay**
## About The Pull Request
<details>
<summary>
	I am once again asking for your coding support 😔 
</summary>
<hr>

- "ports" the entire ATMOSPHERICS folder from eris, adjustments made by me
- air alarms code fixed to make the scrubbers/vents use the correct animations
- meters changed to handle pipe code changes
- re-tgui pumps/passive gates
- average 20 seconds faster server start on my pc
- server locally restarted 15 times, zero "defcons" found
- pipes so far had no clogging issues due to bad pipenet, more testing on live would be ideal

	
<hr>
</details>

## Changelog
:cl:
refactor: ATMOSPHERICS folder
fix: air alarms now order the scrubbers/vents to use the correct animations when changing modes
/:cl:

### CODE REVIEW

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm [hunk 1]
```diff
@@ -75,18 +75,18 @@
 
 /obj/machinery/atmospherics/binary/circulator/update_icon()
 	icon_state = anchored ? "circ-assembled" : "circ-unassembled"
-	cut_overlays()
+	overlays.Cut()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm:78]: Same as before with these kinda things sadly

Comment (Dissentbay) [reply code/ATMOSPHERICS/components/binary_devices/circulator.dm:78]: yeah, but in the omni filters they need to be the way they are or they do not work at all
but yes, on the circulator changing still works

**Comment by Dissentbay** [comment]
also the vents and scrubbers didn't made power in my local test
but I can fix it later easy if they do on the server

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm [hunk 2]
```diff
@@ -80,13 +80,13 @@
 		return
 	if (last_pressure_delta > 0 && recent_moles_transferred > 0)
 		if (temperature_overlay)
-			add_overlay(temperature_overlay)
+			overlays += temperature_overlay
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm:83]: Please make sure to not use the overlay += for overlays as add_overlay is better in every way

File: code/ATMOSPHERICS/components/binary_devices/pipeturbine.dm [hunk 3]
```diff
@@ -73,15 +73,15 @@
 			network2.update = 1
 
 	update_icon()
-		cut_overlays()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/binary_devices/pipeturbine.dm:76]: same with this section

File: code/ATMOSPHERICS/pipes.dm [hunk 4]
```diff
@@ -274,7 +250,7 @@
 
 	alpha = 255
 
-	cut_overlays()
+	overlays.Cut()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/pipes.dm:253]: Sadness

File: code/ATMOSPHERICS/pipes.dm [hunk 5]
```diff
@@ -285,9 +261,9 @@
 				qdel(meter)
 		qdel(src)
 	else if(node1 && node2)
-		add_overlay(icon_manager.get_atmos_icon("pipe", , pipe_color, "[pipe_icon]intact[icon_connect_type]"))
+		overlays += icon_manager.get_atmos_icon("pipe", , pipe_color, "[pipe_icon]intact[icon_connect_type]")
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/pipes.dm:264]: More sadness

File: code/ATMOSPHERICS/pipes.dm [hunk 6]
```diff
@@ -534,9 +510,9 @@
 				qdel(meter)
 		qdel(src)
 	else
-		cut_overlays()
-		add_overlay(icon_manager.get_atmos_icon("manifold", , pipe_color, "core" + icon_connect_type))
-		add_overlay(icon_manager.get_atmos_icon("manifold", , , "clamps" + icon_connect_type))
+		overlays.Cut()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/pipes.dm:513]: All the sadness

File: code/ATMOSPHERICS/pipes.dm [hunk 7]
```diff
@@ -782,9 +758,9 @@
 				qdel(meter)
 		qdel(src)
 	else
-		cut_overlays()
-		add_overlay(icon_manager.get_atmos_icon("manifold", , pipe_color, "4way" + icon_connect_type))
-		add_overlay(icon_manager.get_atmos_icon("manifold", , , "clamps_4way" + icon_connect_type))
+		overlays.Cut()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/pipes.dm:761]: Tons of sadness here as well

File: code/ATMOSPHERICS/pipes.dm [hunk 8]
```diff
@@ -1002,8 +978,8 @@
 
 	alpha = 255
 
-	cut_overlays()
-	add_overlay(icon_manager.get_atmos_icon("pipe", , pipe_color, "cap"))
+	overlays.Cut()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/pipes.dm:981]: Same overlay issue

File: code/ATMOSPHERICS/pipes.dm [hunk 9]
```diff
@@ -1315,8 +1288,8 @@
 
 	alpha = 255
 
-	cut_overlays()
-	add_overlay(icon_manager.get_atmos_icon("pipe", , pipe_color, "universal"))
+	overlays.Cut()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/pipes.dm:1291]: Overlay issues

File: code/ATMOSPHERICS/pipes.dm [hunk 10]
```diff
@@ -1352,8 +1325,8 @@
...
+	overlays += icon_manager.get_atmos_icon("pipe", , pipe_color, "universal")
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/pipes.dm:1329]: as above

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm [hunk 11]
```diff
@@ -82,11 +82,11 @@
 		if (temperature_overlay)
 			add_overlay(temperature_overlay)
 		if (last_pressure_delta > 5*ONE_ATMOSPHERE)
-			add_overlay("circ-run")
+			overlays += "circ-run"
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm:85]: Three more here

File: code/ATMOSPHERICS/components/binary_devices/dp_vent_pump.dm [hunk 12]
```diff
@@ -59,7 +59,7 @@
 	if(!check_icon_cache())
 		return
 
-	cut_overlays()
+	overlays.Cut()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/binary_devices/dp_vent_pump.dm:62]: One here

File: code/ATMOSPHERICS/components/binary_devices/dp_vent_pump.dm [hunk 13]
```diff
@@ -75,7 +75,7 @@
 	else
 		vent_icon += "[use_power ? "[pump_direction ? "out" : "in"]" : "off"]"
 
-	add_overlay(icon_manager.get_atmos_icon("device", , , vent_icon))
+	overlays += icon_manager.get_atmos_icon("device", , , vent_icon)
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/binary_devices/dp_vent_pump.dm:78]: One here

**Comment by Dissentbay** [comment]
jeez, all overlay stuff

**Comment by Trilbyspaceclone** [comment]
No issues has been happening

**Pull request merged**